backend/src/chainArchive.ts

```diff
@@ -40,23 +45,27 @@ class ChainArchive {
   }
 
   public async *getBlocks(lastProcessedBlock: number): AsyncGenerator<ArchivedBlock, void> {
-    this.logger.info('Start processing blocks from archive');
+    try {
+      this.logger.info('Start processing blocks from archive');
 
-    const lastFromDb = await this.getLastBlockNumberFromDb();
-    let nextBlockToProcess = lastProcessedBlock + 1;
+      const lastFromDb = await this.getLastBlockNumberFromDb();
+      let nextBlockToProcess = lastProcessedBlock + 1;
 
-    while (nextBlockToProcess <= lastFromDb) {
-      const blockData = await this.getBlockByNumber(nextBlockToProcess);
-      const blockHashLength = blockData[0];
-      const hash: HexString = `0x${blockData.slice(1, blockHashLength + 1).toString('hex')}`;
-      const blockBytes = blockData.slice(blockHashLength + 1);
+      while (nextBlockToProcess <= lastFromDb) {
+        const blockData = await this.getBlockByNumber(nextBlockToProcess);
+        const blockHashLength = blockData[0];
+        const hash: HexString = `0x${blockData.slice(1, blockHashLength + 1).toString('hex')}`;
+        const blockBytes = blockData.slice(blockHashLength + 1);
 
-      yield new ArchivedBlock(blockBytes, {
-        number: nextBlockToProcess,
-        hash,
-      });
+        yield new ArchivedBlock(blockBytes, {
+          number: nextBlockToProcess,
+          hash,
+        });
 
-      nextBlockToProcess++;
+        nextBlockToProcess++;
+      }
+    } finally {
+      this.db.close();
```

Comment:
We should probably open and close database within the same method. It is very non-intuitive that after calling this method database `ChainArchive` is no longer usable because underlying database is closed.

Alternatively we might provide `ChainArchive.close()` method, but in TypeScript it is impossible to enforce calling it :disappointed:

Reply:
yes, makes sense